src/XRP/default-xrp-client.ts

```diff
@@ -219,7 +219,15 @@ export default class DefaultXrpClient implements XrpClientDecorator {
         .forEach((memo) => transaction.addMemos(memo))
     }
 
-    return this.coreXrplClient.signAndSubmitTransaction(transaction, sender)
+    const transactionHash = await this.coreXrplClient.signAndSubmitTransaction(
+      transaction,
+      sender,
+    )
+    return TransactionResult.getPendingTransactionResult(
```

Comment:
this is a random thought, but now that i'm seeing this, i think this function should be called `createPendingTransactionResult` - i thought get was doing some computation or something

Reply:
I totally agree with that - I renamed this and also its counterpart `getFinalTransactionResult` to start with `create`